src/NUnitFramework/framework/Constraints/EachItemConstraintResult.cs

```diff
@@ -0,0 +1,40 @@
+namespace NUnit.Framework.Constraints
+{
+    /// <summary>
+    /// Provides a <see cref="ConstraintResult"/> for the constraints 
+    /// that are applied to each item in the collection
+    /// </summary>
+    public class EachItemConstraintResult : ConstraintResult
```

Comment:
Could this be an internal class? 🙂

Reply:
(And sealed too, I believe!)

Reply:
Can't see why not.
I tried to make it consistent with other classes, which are public (CollectionEquivalentConstraintResult, EqualConstraintResult)

Reply:
Ah, ok! Historically, lots of classes in NUnit have been public, when actually there was no need for them to be visible to end users. That's unfortunately meant however that people have begun to rely on the behaviour/structure of classes we didn't intend to be public, and made it difficult for us to make changes.

With 'new code' - we're working on only exposing what we need to. 🙂

Reply:
Yes, `internal` or even better IMO, nested within the constraint class and `private`. Makes it super clear how it can be accessed and why.

Reply:
Can't make it nested, as it used both by AllItemsConstraint and NoItemConstraint

Reply:
We don't support C# 7.2 or I'd ask if it makes sense to nest in the base class and make it `private protected`.